docs/decisions/infra/0005-database-migration-architecture.md

```diff
@@ -0,0 +1,129 @@
+
+#### Pros
+This approach uses existing scripts from the application codebase and is quick and simple to set up.
+
+#### Cons
+
+
+### Execute via a Lambda Function
+
+In this method, a Lambda function is used as the primary method of migration. Either the Lambda is the only compute infrastructure, or it works in conjunction with an EC2 instance or ECS task running something like Flyway. The Lambda needs some sort of stateful supporting infrastructure (like S3 or EC2) for saving the state of the migration, as Lambdas are inherently stateless.
+
+#### Infrastructure Required
+
+- Lambda function to run the migration code
+- Log group for storing logs
+- IAM role for the Lambda to use
+- Some other infrastructure to store the state of the migration (S3, EC2, ECS)
+- Any associated IAM roles for the state-storing infrastructure to use
+- Security group that allows connection to the database
+- IAM role that Github can assume to run the task
+- (optional) A queueing method for the Lambda to feed off of, like SQS
+
+#### Pros
+
+Lambda functions can be very cost effective and fast, as well as simple to use.
+
+Lambdas are low maintenance, and don't stick around after the task.
+
+Even though this solution requires additional support, the compute portion of the task run by AWS Lambda is very simple.
+
+#### Cons
+
+By nature, database migrations need to keep track of state--changes between the previous version, the next version, and at what point the migration is currently at. Lambda functions are stateless by design, so an additional component will be needed to keep track of state.
+
+Lambdas have a comparatively short running time. Either find a way to split up the migrations into small tasks, or extend the running time of the Lambda. This might not scale well with large changesets.
+
+
+### Execute via an ECS Fargate Task
+
+In this method, the migration is ran by an ECS task that spins up in a cluster.
+This task will be destroyed once the migration is completed, so it will not need to be
+a long running service.
+
+This method gets the latest configuration by creating a Dockerized image from the
+workflow and pushing it to AWS ECR. This image has a function that will be called
+from the ECS task's configuration JSON.
+
+#### Infrastructure Required
+
+ - ECS cluster to run the task in, which can be the same cluster as the application
+ - Log group for storing logs
+ - IAM role for the task to use
+ - (Optional) IAM policy with permission to connection if using IAM authentication on the database
+ - (Optional) ECR to store the image. This is dependent on the application used
+ - Security group that allows connection to the database
+ - IAM role that GitHub can assume to run the task
+
+#### Pros
+
+Depending on the application, you can overload the ECS task configuration to run
+the migration command and use the same app image, so there is no need to maintain
+a different image. This allows you to run the same code locally as in AWS to try
```

Comment:
> Depending on the application, you can overload the ECS task configuration to run
the migration command and use the same app image, so there is no need to maintain
a different image. 

i thought this is also true of lambdas? can't we configure a lambda to use an existing application image?

Reply:
Good catch, I'll move that to another section saying that both options will use the same Docker image as the app, and that it's a requirement like we discussed in the other thread